### PR TITLE
[FLOC-4008] Have pylint fail on unused variables.

### PR DESCRIPTION
**Pull request by james-w**

### CODE REVIEW

File: flocker/node/test/test_p2p.py [hunk 1]
```diff
@@ -951,7 +951,7 @@ def resized(ignored):
         def got_filesystems(filesystems):
             (filesystem,) = filesystems
             self.assertEqual(size, filesystem.size)
-        d.addCallback(resized)
+        d.addCallback(got_filesystems)
```
Comment (itamarst) [inline flocker/node/test/test_p2p.py]: It caught a bug! :tada:

File: flocker/restapi/docs/test/test_publicapi.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def test_must_be_documented(self):
         app = Klein()
 
         @app.route(b"/", methods=[b"GET"])
-        def f():
+        def dummy_f():
```
Comment (itamarst) [inline flocker/restapi/docs/test/test_publicapi.py]: `dummy_` is pylint-ese for "unused"?

**Comment by itamarst** [comment]
Cool, looks good.

File: flocker/restapi/docs/test/test_publicapi.py (hunk 2 above)
Comment (james-w) [inline flocker/restapi/docs/test/test_publicapi.py]: Yeah, it's in the default config alongside "_" as an unused variable, I
didn't want to define the a function called "_", so used dummy. Open
to changing that.

**Pull request merged**